database/pgx/pgx.go

```diff
@@ -220,7 +221,7 @@ func (p *Postgres) Close() error {
 
 // https://www.postgresql.org/docs/9.6/static/explicit-locking.html#ADVISORY-LOCKS
 func (p *Postgres) Lock() error {
-	if p.isLocked {
+	if p.isLocked.Load() {
```

Comment:
We should replace this with `CAS`, release the local if the postgres lock fails, and remove the `CAS` call below for a proper local lock implementation

Reply:
A lot of thanks for review. fixed it